pkg/runctl/run_manager_test.go

```diff
@@ -505,14 +514,92 @@ func Test_RunManager_setupNetworkPolicyFromConfig_ReplacesAllMetadata(t *testing
 	}
 }
 
-func Test_RunManager_setupNetworkPolicyFromConfig_MalformedPolicy(t *testing.T) {
+func Test_RunManager_setupNetworkPolicyFromConfig_ChooseCorrectPolicy(t *testing.T) {
 	t.Parallel()
 
-	// SETUP
 	const (
-		runNamespaceName = "runNamespace1"
+		expectedNamePrefix = "steward.sap.com--configured-"
 	)
-	runCtx := &runContext{runNamespace: runNamespaceName}
+	for _, tc := range []struct {
+		name           string
+		pipelineSpec   api.PipelineSpec
+		expectedPolicy string
+		expectError    bool
+		result         api.Result
+	}{
+		{"undefined", api.PipelineSpec{}, "networkPolicySpecDefault1", false, api.ResultUndefined},
+		{"nil_profile", api.PipelineSpec{Profiles: nil}, "networkPolicySpecDefault1", false, api.ResultUndefined},
+		{"empty_network_name", api.PipelineSpec{Profiles: &api.Profiles{Network: ""}}, "networkPolicySpecDefault1", false, api.ResultUndefined},
+		{"unknown_network_name", api.PipelineSpec{Profiles: &api.Profiles{Network: "unknown"}}, "", true, api.ResultErrorConfig},
+		{"choose_network_policy1", api.PipelineSpec{Profiles: &api.Profiles{Network: "networkPolicyKey1"}}, "networkPolicySpec1", false, api.ResultUndefined},
+		{"choose_network_policy2", api.PipelineSpec{Profiles: &api.Profiles{Network: "networkPolicyKey2"}}, "networkPolicySpec2", false, api.ResultUndefined},
```

Comment:
This is hardly readably. Better use a style for each test case (using vertical space is a good thing :-) similar to this:

```go
	{
		name: "empty_network_name",
		pipelineSpec: api.PipelineSpec{
			Profiles: &api.Profiles{Network: ""},
		},
		expectedPolicy: "networkPolicySpecDefault1",
		expectError: false,
		result: api.ResultUndefined,
	},
```

- Named fields
- Each field on separate line.
- Non-trivial struct initializers over multiple lines with proper indentation.
- Opening and closing braces at the same column.

Reply:
Done